File: src/main/java/com/destino/projeto_destino/dto/auth/login/LoginResponseDTO.java
package com.destino.projeto_destino.dto.auth.login;

import com.fasterxml.jackson.annotation.JsonInclude;

import java.util.UUID;

@JsonInclude(JsonInclude.Include.NON_NULL)
public record LoginResponseDTO(
        Boolean erro,
        String mensagem,
        UserInfo dados
) {
    public record UserInfo(
            UUID id,
            String nomeCompleto,
            String email,
            String perfil,
            String accessToken,
            Long expiracaoEmSegundos
    ) {
    }
}

File: src/main/kotlin/com/fatec/destino/dto/auth/login/LoginResponseDTO.kt
package com.fatec.destino.dto.auth.login

import com.fasterxml.jackson.annotation.JsonInclude
import java.util.*

@JsonInclude(JsonInclude.Include.NON_NULL)
data class LoginResponseDTO(
    val erro: Boolean?,
    val mensagem: String?,
    val dados: UserInfo?
) {
    data class UserInfo(
        val id: UUID?,
        val nomeCompleto: String?,
        val email: String?,
        val perfil: String?,
        val accessToken: String?,
        val expiracaoEmSegundos: Long?
    )
}
